### PR TITLE
Support converting from bech32 back to hex. Add tests.

### PR DESCRIPTION
**Pull request by KotlinGeekDev**

### CODE REVIEW

**Comment by KotlinGeekDev** [comment]
cc @Giszmo

File: nostrpostrlib/src/main/java/nostr/postr/Utils.kt [hunk 1]
```diff
@@ -81,4 +81,11 @@ fun Int.toByteArray(): ByteArray {
 }
 
 fun ByteArray.toNsec() = Bech32.encodeBytes(hrp = "nsec", this, Bech32.Encoding.Bech32)
-fun ByteArray.toNpub() = Bech32.encodeBytes(hrp = "npub", this, Bech32.Encoding.Bech32)
\ No newline at end of file
+fun ByteArray.toNpub() = Bech32.encodeBytes(hrp = "npub", this, Bech32.Encoding.Bech32)
+
+fun String.rawHexForm() = Bech32.decodeBytes(this).second
```
Comment (Giszmo) [inline nostrpostrlib/src/main/java/nostr/postr/Utils.kt]: Does rawHexForm() return a byte array? Why not name it

```suggestion
fun String.bechToBytes() = Bech32.decodeBytes(this).second
```

or if the hrp is ignored, we could put in some safe-guards by at least making it easy to fail if the user put in his nsec where we expected an npub:

```suggestion
fun String.bechToBytes(hrp:String?=null): ByteArray {
    val decoded = Bech32.decodeBytes(this)
    hrp?.also { 
        if (it !== decoded.first) {
            throw IllegalArgumentException("Expected $it but was ${decoded.first}")
        }
    }
    return decoded.second
}
```

File: nostrpostrlib/src/main/java/nostr/postr/Utils.kt [hunk 2]
```diff
@@ -81,4 +81,11 @@ fun Int.toByteArray(): ByteArray {
...
+
+fun String.rawHexFormWithHrp(): Pair<String, ByteArray> {
+    val decodedHex = Bech32.decodeBytes(this)
+    return Pair(decodedHex.first, decodedHex.second)
+}
```
Comment (Giszmo) [inline nostrpostrlib/src/main/java/nostr/postr/Utils.kt]: Nothing hex-y here neither. With some Kotlin magic I get this one-liner:

```suggestion
/**
 * Interpret the string as Bech32 encoded and return hrp and ByteArray as Pair
 **/
fun String.bechToHrpBytes() = Bech32.decodeBytes(this).run { Pair(first, second) }
```

**Review by Giszmo — COMMENTED** [review]
some ideas

File: nostrpostrlib/src/main/java/nostr/postr/Utils.kt (hunk 1 above)
Comment (KotlinGeekDev) [reply nostrpostrlib/src/main/java/nostr/postr/Utils.kt]: Will adjust it then.:+1:

File: nostrpostrlib/src/main/java/nostr/postr/Utils.kt (hunk 2 above)
Comment (KotlinGeekDev) [reply nostrpostrlib/src/main/java/nostr/postr/Utils.kt]: That's true! I forgot about this.:sweat_smile:

**Comment by KotlinGeekDev** [comment]
Changes made. Ready to merge, @Giszmo .

**Review by Giszmo — APPROVED** [review]

**Comment by Giszmo** [comment]
Thanks!

**Pull request merged**